lib/plugins/digging.js

```diff
@@ -95,6 +102,11 @@ function inject (bot) {
           }
           await bot.lookAt(closest.targetPos, forceLook)
           diggingFace = closest.face
+        } else if (closerBlocks.length === 0 && block.shapes.length === 0) {
+          // no other blocks were detected and the block has no shapes.
+          // The block in question is replaceable (like tall grass) so we can just dig it
+          // TODO: do AABB + ray intercept check to this position for diggingFace.
+          await bot.lookAt(block.position.offset(0.5, 0.5, 0.5), forceLook)
```

Comment:
Ok, my understanding here is if the user doesn't specify a digging direction, mineflayer defaults to doing a raytrace to find the direction to look at before digging.

You want to:
1. prevent ray trace from matching a block closer than the expected target
2. handle raycast failures because the block has no shape, as raycast detection needs shapes to determine a collision by preventing the exception in the else condition (code already defaults to a hardcoded constant)

minecraft-assets should have the block model data, so prismarine-world should be able to refer to that when checking for collisions as opposed to the blockCollisions.json in minecraft-data

But otherwise LGTM for a temporary fix in mineflayer